tests/setup.sh

```diff
@@ -70,5 +70,5 @@ setup_autopin() {
     cp -f "$CODE"/utils/autopin.sh "$EGPG_DIR"/ &&
     local autopin="$EGPG_DIR"/autopin.sh &&
     sed -i "$autopin" -e "/^PIN=/ c PIN='$pin'" &&
-    sed -i "$GNUPGHOME"/gpg-agent.conf -e "/^pinentry-program/ c pinentry-program \"$autopin\""
+    sed -i "$EGPG_DIR"/.gnupg/gpg-agent.conf -e "/^pinentry-program/ c pinentry-program \"$autopin\""
```

Comment:
I think you got this right. Maybe this was the problem that the configuration `gpg-agent.conf` was apparently being ignored.

Reply:
Though this is right it doesn't solves the issue.
As said in previous comments, this helps in getting the configuration right but still it is not working. As said earlier despite correcting this autopin.sh is being launched with sign and seal scripts but for new `passwd` it is launching `pinentry-tty` only. 
I think you might have missed some parts of my comment where I was talking about strange behavior of gpg-agent.conf in which it was launching different programs for new password and other (sign, seal) respectively.

Reply:
> strange behavior of gpg-agent.conf in which it was launching different programs for new password and other (sign, seal) respectively

If you can reproduce and verify this behavior outside of tests, maybe we can ask on gnupg-devel about this issue. But we have to show some steps about how to reproduce this problem.

Reply:
The problem is that we have forgotten to reload the gpg-agent configuration:
`gpgconf --homedir="$homedir" --reload gpg-agent`
So, the changes on the config file do not take effect and the old pinentry is used.
I had tried to reload it before, but without the option `--homedir` (which actually is not so well documented on `gpgconf`). It turns out that there is a different gpg-agent for each homedir, and I was not reloading the right one.

Reply:
@dashohoxha Thanks a lot. This has been bugging me for quite some time. 
I have updated the tests similar to your PR in egpg.

I have updated sign.py also as initially I wasn't showing for bad passphrase.